### PR TITLE
Rails Capstone Project

### PR DESCRIPTION
**Pull request by Lul-Abdifan**
#### Interactions
- Splash screen
  - A simple page with the name of your app (yes, you need to choose one), and links to the sign up and log in pages.

- Sign up and log in pages
  - The user should be able to register in the app with full name, email and password (all mandatory).
  - The user can log into the app using email and password.
  - If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

- Home page (categories page)
  - When the user logs in, they are presented with the categories page.
  - For each category, the user can see their name, icon and the total amount of all the transactions that belongs to that category.
  - When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.
  - There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

- Transactions page
  - For a given category, the list of transactions is presented, ordered by the most recent.
  - At the top of the page the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
  - There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new category" page
  - The user fills out a form to create a new category, indicating their name and icon (both mandatory).
  - The user clicks (or taps) the "Save" button to create the new category, and is taken to the home page on success.
  - When the user clicks on the "Back" button (<), the user navigates to the home page.

- "Add a new transaction" page
  - The user fills out a form to create a new transaction with:
    - name (mandatory)
    - amount (mandatory)
    - categories (mandatory at least one)
  - The user click (or taps) the "Save" button to create the new transaction, and is taken to the transactions page for that category.
  - When the user clicks on the "Back" button (<), the user navigates to the transactions page for that category.

[Demonstration Video](https://drive.google.com/file/d/1XTb9UmYgSXLsCsNnMQNXCedUPp2-Upex/view?usp=sharing)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -45,247 +14,162 @@ For more information on the importance of a professional README for your reposit
...
-  cd my-folder
-  git clone git@github.com:myaccount/my-project.git
+  git clone https://github.com/Lul-Abdifan/Expense-tracker.git
+  cd Expense-tracker
 ```
---->
+### Live Demo
 
+```sh
+[Live](https://expense-tracker-uvao.onrender.com/)
```
Comment (mikethreels) [inline README.md]: - [ ] when opening the live demo the user gets a 502 error, please check and fix this issue before requesting another review.

File: README.md [hunk 2]
```diff
@@ -45,247 +14,162 @@ For more information on the importance of a professional README for your reposit
...
 
 <!-- ACKNOWLEDGEMENTS -->
 
 ## 🙏 Acknowledgments <a name="acknowledgements"></a>
 
-> Give credit to everyone who inspired your codebase.
+I would like to thanks Microverse for this learning opportunity.
 
-I would like to thank...
+I would like also The [Creative Commons license of the design](https://creativecommons.org/licenses/by-nc/4.0/) for incredible design
```
Comment (mikethreels) [inline README.md]: - [ ] please make sure to name the creator [Gregoire Vella on Behance](https://www.behance.net/gregoirevella)  in your acknowledgements.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Hope, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- good job setting up the routes.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by KDlamini — APPROVED** [review]
### Hi @Lul-Abdifan 🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes for the Budget App capstone. 🎊 🥳 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## Highlights 🌞 

✔️  Professional pull request
✔️  All required pages for categories, transactions, and user authentications are well-implemented
✔️  Database setup, controllers, models, routes, and views implemented correctly
✔️  The app schema looks similar to the Entity Relationship Diagram
✔️  The design guidelines are followed
✔️  App uses RSpec and Capybara testing libraries

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by Lul-Abdifan** [comment]
@KDlamini  I appreciate your feedback bro. Thank you

**Pull request closed without merging**